Script/AtomicNET/AtomicNETService/CSComponentInspector.cs

```diff
@@ -34,721 +34,796 @@
+        {
+            return spans != null && spanIndex < spans.Count && spans[spanIndex].StartOffset == (uint)curIndex;
+        }
+
+        private static bool EndsSpan(IReadOnlyList<HandlerSpan> spans, int spanIndex, int curIndex)
+        {
+            return spans != null && spanIndex >= 0 && spans[spanIndex].EndOffset == (uint)curIndex;
+        }
+
+        private static bool StartsFilterHandler(IReadOnlyList<HandlerSpan> spans, int spanIndex, int curIndex)
+        {
+            return spans != null &&
+            spanIndex < spans.Count &&
+            spans[spanIndex].Kind == HandlerKind.Filter &&
+            spans[spanIndex].FilterHandlerStart == (uint)curIndex;
+        }
+
+        public static IReadOnlyList<HandlerSpan> GetHandlerSpans(ImmutableArray<ExceptionRegion> entries)
+        {
+            if (entries.Length == 0)
+            {
+                return new HandlerSpan[0];
+            }
+
+            var result = new List<HandlerSpan>();
+            foreach (ExceptionRegion entry in entries)
+            {
+                int tryStartOffset = entry.TryOffset;
+                int tryEndOffset = entry.TryOffset + entry.TryLength;
+                var span = new HandlerSpan(HandlerKind.Try, null, tryStartOffset, tryEndOffset);
+
+                if (result.Count == 0 || span.CompareTo(result[result.Count - 1]) != 0)
+                {
+                    result.Add(span);
+                }
+            }
+
+            foreach (ExceptionRegion entry in entries)
+            {
+                int handlerStartOffset = entry.HandlerOffset;
+                int handlerEndOffset = entry.HandlerOffset + entry.HandlerLength;
+
+                HandlerSpan span;
+                switch (entry.Kind)
+                {
+                    case ExceptionRegionKind.Catch:
+                        span = new HandlerSpan(HandlerKind.Catch, MetadataTokens.GetToken(entry.CatchType), handlerStartOffset, handlerEndOffset);
+                        break;
+
+                    case ExceptionRegionKind.Fault:
+                        span = new HandlerSpan(HandlerKind.Fault, null, handlerStartOffset, handlerEndOffset);
+                        break;
+
+                    case ExceptionRegionKind.Filter:
+                        span = new HandlerSpan(HandlerKind.Filter, null, handlerStartOffset, handlerEndOffset, entry.FilterOffset);
+                        break;
+
+                    case ExceptionRegionKind.Finally:
+                        span = new HandlerSpan(HandlerKind.Finally, null, handlerStartOffset, handlerEndOffset);
+                        break;
+
+                    default:
+                        throw new InvalidOperationException();
+                }
+
+                result.Add(span);
+            }
+
+            return result;
+        }
+
+        public void Dump()
+        {
+            /*
 				foreach (var entry in InspectorFields) {
```

Comment:
Ok, this may be an issue when inheriting components across assemblies, instead of needing to be able to gather full information during the assembly parse, it may be necessary/better to combine the inspector fields at edit time, we'll cross that bridge when we get to it :)